ElectricCar.cpp

```diff
@@ -7,10 +7,23 @@ ElectricCar::ElectricCar(ElectricEngine* engine)
     std::cout << __FUNCTION__ << std::endl;
 }
 
-ElectricCar::~ElectricCar()       { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
+ElectricCar::~ElectricCar() {
+    delete engine_;
+    std::cout << __FUNCTION__ << std::endl;
+}
+void ElectricCar::restore()       { charge(); }
 void ElectricCar::charge()        { std::cout << __FUNCTION__ << std::endl; }
+void ElectricCar::setEngine(int power, int batteryCapacity) {
+    if (Car::speed_ == 0) {
```

Comment:
Maybe it would be better to throw an exception?

Reply:
done